### ISSUE TITLE
headErr and tailErr 

### ISSUE THREAD

**Issue by rd--** [ndmitchell/safe#34]
Ghc 9.8.1 prints warnings for head and tail so I'm wondering if this library might be a nice place to have headErr and tailErr, for cases where the traditional (perfectly well defined and documented!) head and tail definitions are the ones that one actually wants?  I.e.

```
headErr             :: [a] -> a
headErr (x:_)       =  x
headErr []          =  error "Safe.headErr: empty list"

tailErr             :: [a] -> [a]
tailErr (_:xs)      =  xs
tailErr []          =  error "Safe.tailErr: empty list"
```

**Comment by ndmitchell** [comment]
That seems very reasonable. Want to send a PR?

**Comment by ndmitchell** [comment]
I'm going to make a release pretty much now, so added them myself.

**Comment by adamgundry** [comment]
Wouldn't it make sense to add `Partial` (`HasCallStack`) constraints to these functions, so they give useful call stacks when the error is thrown?

**Comment by ndmitchell** [comment]
@adamgundry good point - added Partial and released 0.3.21
